### PR TITLE
Add a flow to run tests every 2 hours, report results to slack

### PR DESCRIPTION
**Pull request by amurKontur**
## Summary by CodeRabbit

- **New Features**
  - Introduced Slack reporting for end-to-end test results.
  - Added user management features in Keycloak, including registration and email verification.

- **Bug Fixes**
  - Updated tests to exclude projects without necessary features, improving test accuracy.

- **Enhancements**
  - Improved artifact upload process and added report uploading to `surge.sh`.
  - Enhanced toolbar feature checks based on project type.

- **Chores**
  - Updated environment variables and dependencies in configuration files.
  - Refined test setup for better project filtering and logic handling.

### CODE REVIEW

File: .github/workflows/run_e2e_tests.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 name: Run e2e autotests
+run-name: Run e2e autotests for ${{inputs.app || 'all'}} app(s) on ${{inputs.env || 'prod'}} env
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply .github/workflows/run_e2e_tests.yml:2]: I suppose it is obvious from the workflow code

File: .github/workflows/run_e2e_tests.yml [hunk 2]
```diff
@@ -19,21 +20,29 @@
...
       TEST_PASSWORD: ${{ secrets.TEST_PASSWORD }}
       DEV_EMAIL: ${{ secrets.DEV_EMAIL }}
       DEV_PASSWORD: ${{ secrets.DEV_PASSWORD }}
       PROD_EMAIL: ${{ secrets.PROD_EMAIL }}
       PROD_PASSWORD: ${{ secrets.PROD_PASSWORD }}
+      TEST_KEYCLOAK_ADMIN_USER: ${{ secrets.TEST_KEYCLOAK_ADMIN_USER }}
+      TEST_KEYCLOAK_ADMIN_PASSWORD: ${{ secrets.TEST_KEYCLOAK_ADMIN_PASSWORD }}
+      DEV_KEYCLOAK_ADMIN_USER: ${{ secrets.DEV_KEYCLOAK_ADMIN_USER }}
+      DEV_KEYCLOAK_ADMIN_PASSWORD: ${{ secrets.DEV_KEYCLOAK_ADMIN_PASSWORD }}
+      SLACK_BOT_USER_OAUTH_TOKEN: ${{secrets.PLAYWRIGHT_TOKEN }}
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply .github/workflows/run_e2e_tests.yml]: all of that is already done

File: e2e/page-objects/keycloakPage.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        password: adminPassword,
+        client_id: 'security-admin-console',
+        scope: 'openid',
+      },
+    });
+
+    expect(response.status()).toEqual(200);
+    const responseBody = await response.json();
+    return responseBody.access_token;
+  }
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/keycloakPage.ts]: they are not logged

File: playwright.config.ts [hunk 4]
```diff
@@ -28,7 +29,53 @@ export default defineConfig({
   /* Opt out of parallel tests on CI. */
   workers: process.env.CI ? 4 : 6,
   /* Reporter to use. See https://playwright.dev/docs/test-reporters */
-  reporter: 'html',
+  reporter: [
+    [
+      './node_modules/playwright-slack-report/dist/src/SlackReporter.js',
```
Comment (gabriel-fallen) [inline playwright.config.ts:34]: Is it OK to hardcode a path into `node_modules`? Can't it be `import`ed? 🤔

File: playwright.config.ts [hunk 5]
```diff
@@ -28,7 +29,53 @@ export default defineConfig({
...
+        sendResults: 'always', // "always" , "on-failure", "off"
+        maxNumberOfFailuresToShow: 100,
+        meta: [
+          {
+            key: `Tests launched 🎬`,
+            value: `${process.env.CI ? 'Github Actions 🐌' : 'Locally'}`,
+          },
+          {
+            key: `Environment 👷🏻‍♂️`,
+            value: `${process.env.ENVIRONMENT?.toUpperCase()}`,
```
Comment (gabriel-fallen) [inline playwright.config.ts:46]: Maybe
```suggestion
            value: process.env.ENVIRONMENT?.toUpperCase() ?? '',
```
will be better?

**Review by dqunbp — APPROVED** [review]

File: playwright.config.ts (hunk 4 above)
Comment (amurKontur) [reply playwright.config.ts:34]: it is how library works from official documents 

suppose i can change that in future if needed

File: playwright.config.ts (hunk 5 above)
Comment (amurKontur) [reply playwright.config.ts:46]: actually this variable can not be undefined at the report sending step.check for process.env.ENVIRONMENT is added just to make TS calmer when calling toUpperCase() method. It is ok if we fall if env is somehow not set as all tests use this variable behind the scenes. So no, it makes no sense

**Pull request merged**